src/main/java/com/conveyal/datatools/manager/models/transform/NormalizeFieldTransformation.java

```diff
@@ -193,77 +195,111 @@ public void transform(FeedTransformZipTarget zipTarget, MonitorableJob.Status st
-                String transformedValue = originalValue;
+            writeCsvContent(zipTarget, tempZipPath, stringWriter, tableName, modifiedRowCount);
 
-                // Convert to title case, if requested.
-                if (capitalize) {
-                    if (capitalizationStyle == CapitalizationStyle.TITLE_CASE) {
-                        transformedValue = convertToTitleCase(transformedValue);
-                    }
-                    // TODO: Implement other capitalization styles.
-                }
+            // Replace original zip file with temporary working zip file.
+            // (This should also trigger a system IO update event, so subsequent IO calls pick up the correct file.
+            Files.move(tempZipPath, originalZipPath, StandardCopyOption.REPLACE_EXISTING);
+            LOG.info("Field normalization transformation successful, {} row(s) changed.", modifiedRowCount);
+        } catch (ZipException ze) {
+            status.fail(
+                String.format("'Normalize Field' failed because the GTFS archive is corrupted (%s).", ze.getMessage()),
+                ze
+            );
+        } catch (Exception e) {
+            status.fail("Unknown error encountered while transforming zip file", e);
+        }
+    }
 
-                // Perform substitutions if any.
-                transformedValue = performSubstitutions(transformedValue);
+    /** Write csv input stream into the zip file, replacing the existing file. */
+    private void writeCsvContent(
+        FeedTransformZipTarget zipTarget,
+        Path tempZipPath,
+        StringWriter stringWriter,
+        String tableName,
+        int modifiedRowCount
+    ) throws IOException {
+        try (
+            // Modify target zip file that we just read.
+            FileSystem targetZipFs = FileSystems.newFileSystem(tempZipPath, (ClassLoader) null);
+            // Stream for file copy operation.
+            InputStream inputStream = new ByteArrayInputStream(stringWriter.toString().getBytes(StandardCharsets.UTF_8))
+        ) {
+            Path targetTxtFilePath = getTablePathInZip(tableName, targetZipFs);
+            Files.copy(inputStream, targetTxtFilePath, StandardCopyOption.REPLACE_EXISTING);
+            zipTarget.feedTransformResult.tableTransformResults.add(
+                new TableTransformResult(tableName, 0, modifiedRowCount, 0)
+            );
+        }
+    }
 
-                // Re-assemble the CSV line and place in buffer.
-                String[] csvValues = csvReader.getValues();
+    /** Generates headers and content for the GTFS table, returns the number of rows modified. */
+    private int generateCsvContent(CsvListWriter writer, String[] headers, CsvReader csvReader, int transformFieldIndex) throws IOException {
+        int modifiedRowCount = 0;
+
+        // Write headers.
+        // If the index is -1, this is a new column, and we need to add it accordingly.
```

Comment:
Just confirming that all this refactoring is only to add support for adding a new column via the normalize transformation? I have slight worries about how this is going to interact with flex (where I think we solved this problem in a different way) but I guess that's a problem for later

Reply:
The `NormalizeFieldTransformation` logic in the dev branch already adds a new column if a field is unknown, See previous line 201 in the diff. We changed that for MTC in #617.